packages/block-editor/src/components/editor-fonts-resolver/index.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * WordPress dependencies
+ */
+import { useState, useEffect, useMemo, useCallback } from '@wordpress/element';
+import { useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { useGlobalSetting } from '../global-styles/hooks';
+import { getDisplaySrcFromFontFace, loadFontFaceInBrowser } from './utils';
+
+function EditorFontsResolver() {
+	const [ loadedFontUrls, setLoadedFontUrls ] = useState( new Set() );
+
+	const { currentTheme } = useSelect( ( select ) => {
+		return {
+			currentTheme:
+				// Disable Reason: Using 'core' as string to avoid circular dependency importing from @wordpress/core-data.
+				// eslint-disable-next-line @wordpress/data-no-store-string-literals
+				select( 'core' ).getCurrentTheme(),
+		};
+	}, [] );
+
+	// Get the fonts from merged theme.json settings.fontFamilies.
+	const [ fontFamilies = [] ] = useGlobalSetting( 'typography.fontFamilies' );
```

Comment:
Thanks for the ping 👍 

> a new block editor setting that allows us to access global styles in all editors

This was specifically related to the **_styles_** data from Global Styles as the settings values were already included within the block editor settings. 

The Global Styles settings and styles weren't merged as there wasn't consensus on what the shape of those settings should be while maintaining BC. That led to [using a symbol as a private key](https://github.com/WordPress/gutenberg/pull/61556/files#diff-75d4f4d6d3bd43a48a5764edf03d97a190d807658245ff7a8d21cf197222eed3R1) for the Global Styles style data.

Here's the PR adding the style data to the block editor settings: https://github.com/WordPress/gutenberg/pull/61556.

Without looking into the code I'm not sure whether the Global Styles settings under `__experimentalFeatures` in the editor settings are kept in sync with unsaved Global Styles changes in the site editor. It could be possible that aspect needs addressing or perhaps in the short term this could fallback to the editor settings if the Global Styles context isn't available (in post editor).

Reply:
Good point, I mistakenly thought this is about the styles. 

This PR reminds of the "styles" prop that we pass to the block editor to render the current theme styles. I could suggest a new "type" within the same "styles" prop to load fonts (like we do for svg for instance) but first. I would like to understand more.

What fonts are we trying to resolve here. I'm guessing the fonts that are used in the theme styles right? In which case, I feel like "resolving them" or "rendering them" should be done in the [EditorStyles](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/editor-styles/index.js) component. It's a component that is already used to "resolve or render" SVGs and CSS, it's already rendered in both iframe and non-iframe, so I'm starting to think that it's probably the right place for this change? (Introduce a new "type" of style and render it within that component)

Please correct me if my argumentation or abstraction seem too far of

Reply:
> I'm not sure about that, EditorStyles gets styles and not settings. At least, as it is today, that component is not getting the data needed to implement the desired functionality.

EditorStyles renders the CSS variables for the color presets for instance, renders the SVG filters used by Duotone. Isn't it the same for fonts here? We're resolving the fonts that are used in the CSS.

It is possible that I'm misunderstanding the issue that this PR solves though and if it's the case please clarify? I'm assuming that resolving the fonts is done in the frontend somehow right? How is it done today?

> 

The styles prop received by the `EditorStyles` component can contain things like that

```css
const styles = [
	{
		css: `some css`,
	},
	{
                __unstableType: 'svgs'
                assets: `some sig element`
	},
	{
                __unstableType: 'presets'
                assets: `css with CSS variables generated from the presets`
	},
];
```

This represents all what's necessary for the "styles" to be rendered properly and to work properly.
Would be out of the question to consider "fonts" at the same level as these and just add

```
{
              __unstableType: 'fonts'
              fonts: `whatever is needed to resolve the fonts`
},
```

Again, it's possible that I'm wrong

Reply:
> could you show me how is this done? I haven't found how to access the fontFamilies path mentioned before.

In the `block-editor` store, under `settings.__experimentalFeatures.typography.fontFamilies`, you should find the data you're chasing in the post editor.

One example of accessing theme.json settings via the block editor store can be found in the [layout block support](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/hooks/layout.js?plain=1#L81). Another example (that might change soon) is the [block style variations block support](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/hooks/block-style-variation.js?plain=1#L259-L266) which prefers the GlobalStyleContext data, if available, falling back to the block editor store's data.

Hope that helps a little 🤞

Reply:
Thanks for the insights.
In this commit I removed the use `useGlobalSetting` and replaced it by the use data from block-editor store and move the call to the font faces resolved inside the EditorStyles component: https://github.com/WordPress/gutenberg/pull/65019/commits/c6a32f9d7381e4f93a3c471de7214023f4984edd